### PR TITLE
Fixes #6608 - expose disk usage info through available_storage_domains API

### PR DESCRIPTION
**Pull request by gpetras**
Extends the existing available_storage_domains API by adding the freespace, capacity, and uncommitted values.

### CODE REVIEW

**Comment by ohadlevy** [comment]
[test]

**Comment by lzap** [comment]
Thanks @gpetras 

Tests does not work, can you take a look please?

**Comment by gpetras** [comment]
Looks like a few tests were using ovirt as their virtualization implementation and should have been using vmware. I updated those tests and it looks good now. There are still errors when running all of the tests, but those seem to have been present prior to my changes.

**Comment by lzap** [comment]
Thanks, if you can squash all the commits into one:

```
fixes #6608 - expose disk usage....
```

[test]

**Comment by gpetras** [comment]
Hey @lzap, should be good now.

Thanks!

Greg

**Comment by lzap** [comment]
Cool, I will start with the review soon [test]

**Comment by lzap** [comment]
LGTM but untested, I don't have a handy VmWare instance.

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 1]
```diff
@@ -63,8 +71,8 @@ def available_networks(cluster_id=nil)
       networks
     end
 
-    def available_storage_domains
-      datastores
+    def available_storage_domains(storage_domain=nil)
```
Comment (domcleal) [inline app/models/compute_resources/foreman/model/vmware.rb:74]: This method's also defined in ovirt.rb, but it'll need this additional parameter adding or it's going to cause errors.

File: test/functional/api/v2/compute_resources_controller_test.rb [hunk 2]
```diff
@@ -183,7 +183,7 @@ def teardown
 
     Foreman::Model::Vmware.any_instance.stubs(:available_storage_domains).returns([storage_domain])
```
Comment (domcleal) [inline test/functional/api/v2/compute_resources_controller_test.rb]: Could you try changing this to: `Foreman::Model::Vmware.any_instance.expects(:available_storage_domains).with(nil).returns([storage_domain])` as it's now passing an argument in.

**Comment by domcleal** [comment]
Tests fine, but it does break the equivalent oVirt API as noted above.  It'd be great too if a functional test could hit this route - copy "should get available vmware storage domains" and add the storage domain parameter, then using `expects(:available_storage_domains).with('name of domain')` should check it calls the right thing.

File: app/models/compute_resources/foreman/model/vmware.rb (hunk 1 above)
Comment (domcleal) [inline app/models/compute_resources/foreman/model/vmware.rb:74]: Ah, and also in compute_resource.rb.

**Comment by domcleal** [comment]
Merged as a52580f for Foreman 1.7, thanks @gpetras.

**Pull request closed without merging**